Issue-3479: Refactor: Fetch Platform Version

jamrok opened the pull request:
Hey @tjtelan,

I added a common library (`fluvio-common`) to parse the `VERSION` file and made all relevant crates refer to it. I saw that there was `fluvio-cli-common`, but It seemed more specific to `fluvio-cli` versus being a light dependency that can be used by all. 

Please let me know if there's another place that you recommend that the change be made.

Thanks

---
Resolves #3479

tjtelan commented:
@jamrok Thanks for the quick turnaround!

I think there are 2 points of feedback:

1. Your implementation looks good, however not quite the direction I had in mind. The desired outcome is that compilation should be prevented upon discovering errors with parsing `VERSION`, instead of allowing the condition to be found in runtime. I think validation of `VERSION` should be implemented within a `build.rs` build script, in a similar fashion to how we include the current git sha of builds. 

2. `FLUVIO_PLATFORM_VERSION` should live in the `fluvio-types` crate for the purposes of sharing with other crates

jamrok commented:
Thanks for the feedback @tjtelan!

1. Updated as requested. Please also confirm that the `build.rs` for `fluvio-test-util` is done as expected.
    - I moved the compile time check for the optional `FLUVIO_VERSION_SUFFIX` env var to `build.rs` and set a different ENV variable (`FLUVIO_PLATFORM_TEST_VERSION`) so it doesn't conflict with the `FLUVIO_PLATFORM_VERSION` variable.
2. Updated as requested. I originally had `FLUVIO_PLATFORM_VERSION` in `fluvio-types`, but then moved it because `fluvio-types` was published and I was thinking that "types" didn't seem like the right place to put it.
   - Are there any implications with it being published?
   - Should it be placed in `fluvio_types::defaults` and exported at the crate level (`fluvio_types::FLUVIO_PLATFORM_VERSION`), or placed somewhere else?

 Please let me know if you have any other suggestions.

Thanks

jamrok commented:
Hey @tjtelan, just checking on this. I had a couple questions in my last comment.

I also rebased and resolved the merge conflicts.

sehz commented on crates/fluvio-types/build.rs:
```diff
@@ -0,0 +1,17 @@
+    println!("cargo:rerun-if-changed=build.rs");
+
+    // Parse the Fluvio Platform Version
+    let version = include_str!("../../VERSION").trim();
```
Why parsing this again? This won't work since it is public crate

tjtelan replied:
I think this is my bad. At the time of [my feedback](https://github.com/infinyon/fluvio/pull/3485#issuecomment-1690070729) I didn't consider that this was a public crate
